### DISCUSSION TITLE
Adding conda and homebrew PATH to .bash profile.

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#24920] [Q&A General] [answered]
Hi, I had a similar problem as discussed in #24342. Although I fixed the error temporarily, I’m wondering how to fix it permanently by adding the PATH to my .bash profile. I can’t figure out where my .bash profile is located. Could someone please direct me? Thanks.

**Comment by milljm** [comment, marked as answer]
Depends on the system... Homebrew is _usually_ a Macintosh user thing, so what you want to change might actually be your ZSH profile. In either case, if you are using Bash these files (and others) are directly located in your `$HOME` directory:

(some files may not exist)
```bash
~/.bashrc         # BASH 
~/.bash_profile   # BASH
~/.profile        # BASH I've seen Homebrew stick things here sometimes
~/.zshrc          # Zsh
```

Why so many BASH profiles? I think the [`man` page sums it up best](https://www.man7.org/linux/man-pages/man1/bash.1.html) (scroll down to "INVOCATION").

Hopefully you are using `Zsh`. As its a bit less complicated. You can verify what shell your terminal defaults to by running the following command (my machine spits back `-zsh`):

```bash
❯ echo $0
-zsh
```

If you are using a GUI file manager, these files (and others which begin with a period) are hidden. They are also likely hidden without using additional listing arguments to `ls`. Example:

```bash
❯ mkdir ~/temp; cd ~/temp
❯ touch .hidden
❯ ls

❯ ls -a
.       ..      .hidden
```

**Reply by steam-simulator** [reply]
thanks, @milljm!
